backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java

```diff
@@ -1567,6 +1573,20 @@ public final boolean isDownPayment(final org.compiere.model.I_C_Invoice invoiceR
 		return docTypeBL.isDownPayment(docTypeId);
 	}
 
+	@Override
+	public final boolean isFinalInvoiceOrFinalCreditMemo(final org.compiere.model.I_C_Invoice invoiceRecord)
+	{
+		final DocTypeId docTypeId = assumeNotNull(getDocTypeIdEffectiveOrNull(invoiceRecord), "The given C_Invoice={} needs to have a C_DocType", invoiceRecord);
```

Comment:
why do we need to have a doctype?
if we don't then for sure is not a final invoice ...

Reply:
Copied logic from other existing checks, but I can also return false then

Reply:
yes pls